src/utils/configUtils.ts

```diff
@@ -103,3 +135,149 @@ async function askPreferenceForConfig(configs: IExecutionConfig[], selectedConfi
 export function randomSequence(): string {
     return crypto.randomBytes(3).toString('hex');
 }
+
+type ApplyOperator = (value1: unknown, value2?: unknown) => boolean
+
+interface Token {
+    stringValue: string
+    getValue: () => unknown
+}
+
+interface ResultToken extends Token {
+    getValue: () => boolean
+}
+
+export class WhenClauseEvaluationContext {
+
+    private static readonly OPERATORS: Record<string, ApplyOperator> = {
+        // logical
+        '!': (value: unknown) => !value,
+        '&&': (value1: unknown, value2: unknown) => !!(value1 && value2),
+        '||': (value1: unknown, value2: unknown) => !!(value1 || value2),
+
+        // equality
+        '==': (value1: unknown, value2: unknown) => value1 === value2,
+        '===': (value1: unknown, value2: unknown) => value1 === value2,
+        '!=': (value1: unknown, value2: unknown) => value1 !== value2,
+        '!==': (value1: unknown, value2: unknown) => value1 !== value2,
+
+        // comparison
+        '>': (value1: number, value2: number) => value1 > value2,
+        '>=': (value1: number, value2: number) => value1 >= value2,
+        '<': (value1: number, value2: number) => value1 < value2,
+        '<=': (value1: number, value2: number) => value1 <= value2,
+
+        // match
+        '=~': (value: string, pattern: RegExp) => pattern.test(value),
+    }
+
+    private readonly context: Record<string, unknown> = {};
+
+    public constructor(readonly clause: string) {}
+
+    private tokenize(): Token[] {
+        const tokens: string[] = this.clause.split(/\s+/)
```

Comment:
Do you want to support clause without space, like `testItem=~xxx`?

Reply:
From https://code.visualstudio.com/api/references/when-clause-contexts#comparison-operators:

> You can compare a context key's value against a number. Note that left- and right-hand side of the operator must be separated by whitespace - `foo < 1`, but not `foo<1`.

It doesn't specify if this is the case for all operators, but all examples given point to that. If not, we could split on all operator keys (and retain them):

```js
        const operatorKeys: string[] = Object.keys(WhenClauseEvaluationContext.OPERATORS);
        const operatorPattern: RegExp = new RegExp(`(${operators.map(_.escapeRegExp).join('|')}`);

        const tokens: string[] = this.clause.split(operatorPattern)
            .flatMap((token: string) => token.split(/([\(\)]|!(?!=))/))
            .filter(Boolean);
```

Reply:
I see. Let's use whitespace as the separator then.

Reply:
Sorry, edited my comment above after you had replied. Any other thoughts?

Reply:
FWIW, I tested the above an it works. Can revert https://github.com/microsoft/vscode-java-test/pull/1638/commits/5c6135e89fa896f168cc58d08a22b71fb5fd669b if not required.

Reply:
It looks good! No need to revert 👍